phac_aspc/django/excel.py

```diff
@@ -312,17 +330,52 @@ def get_sheet_name(self):
 
         return self.sheet_name
 
+    def get_header_style(self):
+        """
+        To be overwritten by child classes when a style is to be applied to the
+        header row.  If overwritten, should return a Style object.
+        """
+        return None
+
+    def get_column_widths(self):
+        """
+        Returns an array of column widths as set in the get_column_configs.
+        """
+        return [col.get_column_width() for col in self.get_column_configs()]
+
     def write(self):
         worksheet = self.workbook.create_sheet(title=self.get_sheet_name())
 
-        worksheet.append(self.get_header_row())
+        for col_index, column_width in enumerate(self.get_column_widths()):
+            if column_width:
```

Comment:
I am tweaking this to an explicit None-check because a column width of 0 would be "falsey" but may still be desired

Reply:
That's a great call.  0-width columns would definitely be useful.